server/routes/archiveGoal/archiveGoalController.test.ts

```diff
@@ -0,0 +1,258 @@
+import { NextFunction, Request, Response } from 'express'
+import type { SessionData } from 'express-session'
+import type { ActionPlan } from 'viewModels'
+import type { ArchiveGoalForm } from 'forms'
+import createError from 'http-errors'
+import type { ArchiveGoalDto } from 'dto'
+import EducationAndWorkPlanService from '../../services/educationAndWorkPlanService'
+import aValidPrisonerSummary from '../../testsupport/prisonerSummaryTestDataBuilder'
+import ArchiveGoalController from './archiveGoalController'
+
+import { aValidActionPlanWithOneGoal, aValidGoal, aValidStep } from '../../testsupport/actionPlanTestDataBuilder'
+import validateArchiveGoalForm from './archiveGoalFormValidator'
+import aValidArchiveGoalForm from '../../testsupport/archiveGoalFormTestDataBuilder'
+import ReasonToArchiveGoalValue from '../../enums/ReasonToArchiveGoalValue'
+import toArchiveGoalDto from './mappers/archiveGoalFormToDtoMapper'
+
+jest.mock('./archiveGoalFormValidator')
+jest.mock('./mappers/archiveGoalFormToDtoMapper')
+
+describe('archiveGoalController', () => {
+  const educationAndWorkPlanService = {
+    getActionPlan: jest.fn(),
+    archiveGoal: jest.fn(),
+  }
+  const mockedValidateArchiveGoalForm = validateArchiveGoalForm as jest.MockedFunction<typeof validateArchiveGoalForm>
+  const mockedArchiveGoalFormToArchiveGoalDtoMapper = toArchiveGoalDto as jest.MockedFunction<typeof toArchiveGoalDto>
+  const controller = new ArchiveGoalController(educationAndWorkPlanService as unknown as EducationAndWorkPlanService)
+
+  const req = {
+    session: {} as SessionData,
+    body: {},
+    user: {} as Express.User,
+    params: {} as Record<string, string>,
+  }
+  const res = {
+    redirect: jest.fn(),
+    redirectWithErrors: jest.fn(),
+    render: jest.fn(),
+  }
+  const next = jest.fn()
+
+  const prisonNumber = 'A1234GC'
+  const goalReference = '1a2eae63-8102-4155-97cb-43d8fb739caf'
+  const prisonerSummary = aValidPrisonerSummary(prisonNumber, 'BXI')
+  let errors: Array<Record<string, string>>
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    req.session = {} as SessionData
+    req.body = {}
+    req.user = {} as Express.User
+    req.params = {} as Record<string, string>
+
+    req.params.prisonNumber = prisonNumber
+    req.params.goalReference = goalReference
+    req.session.prisonerSummary = prisonerSummary
+
+    errors = []
+  })
+  describe('getArchiveGoalView', () => {
+    it('Should load the page with default archive goal form for the reference', async () => {
```

Comment:
Very minor, but the method name including `with default archive goal form` would suggest there is a case/code path where you can get a non-default form in some way?